working/augmentation-libraries/feature-specification.md

```diff
@@ -622,311 +662,89 @@ It is a compile-time error if:
-    or is that too weird?)**
-
-*   An `abstract` variable is augmented with a non-abstract variable.
-
-*   An `external` variable is augmented with an `abstract` variable.
-
-### Augmenting enum members
-
-Some enum members can not be augmented: It is a compile-time error if an
-augmenting declaration in an enum declaration (introductory or augmenting)
-has the name `values`, `index`, `hashCode`, or `==`.
-
-*It has always been an error for an enum declaration to declare a member
-named `index`, `hashCode`, `==`, or `values`, and this rule just clarifies
-that this error is applicable for augmenting declarations as well.*
-
-Enum values can _only_ be augmented by enum values, and the implicit getter
-introduced by them is not augmentable. The only thing you are allowed to do
-when augmenting an enum value is add metadata annotations or doc comments.
+*   A function is not complete after all augmentations are applied, unless it
+    is in a context where it can be abstract. *Every function declaration
+    eventually needs to have a body filled in unless it's an instance method
+    that can be abstract. In that case, if no declaration provides a body, it
+    is considered abstract.*
 
-When augmenting an enum value, no constructor invocation should be provided.
-The original value is always used, and the explicit constructor invocation (if
-present) should not be copied.
+### Augmenting variables
 
-New enum values may be defined in an augmenting enum, and they will be appended
-to the current values of the declaration in augmentation application order.
+A class-like augmentation can add *new* variables (that is instance or static
+fields) to the type being augmented:
 
-Augmenting an existing enum value never changes the order in which it appears in
-`values`.
+```dart
+class C {}
 
-For example:
+augment class C {
+  int x = 3;
 
+  static int y = 4;
+}
 ```
-// main.dart
-part 'a.dart';
-part 'c.dart';
-
-enum A {
-  first,
-  second.custom(1);
 
-  final int b;
-
-  const A() : b = 0;
-
-  const A.custom(this.b);
-}
-}
+Variable declarations themselves can't be augmented (by variables, getters, or
+setters), with the exception of abstract instance variable declarations.
 
-// a.dart
-part of 'main.dart';
-part 'b.dart';
+*A variable declaration implicitly has code for the synthesized getter and
+setter that access and modify the underlying backing storage. Since we don't
+allow augmentations to replace code, that implies that augmentations can't
+change variables. So we don't allow them to be augmented.*
 
-augment enum A {
-  third;
+#### Abstract instance variables
 
-  /// Some doc comment
-  augment first; // This is still `first` in values.
+Dart supports `abstract` field declarations. They are syntactic sugar for
+declaring an abstract getter with an optional abstract setter if the variable is
+non-final. They don't actually declare a variable with any backing storage.
```

Comment:
The word 'optional' is confusing here. Perhaps:
```suggestion
Dart supports `abstract` field declarations. They are syntactic sugar for
declaring an abstract getter and, if not `final`, an abstract setter.
They don't actually declare a variable with any backing storage.
```

Reply:
Ooh, much better. Applied locally.